projects/frontend/src/models/sort-rules.ts

```diff
@@ -2,7 +2,7 @@ export default interface SortRules {
   favorites: boolean;
   small: boolean;
   searchTerm: string,
-  sortOrder: string,
-  groups: string;
+  sortOrder: (string & {}) | 'A-Z' | 'Date - New' | 'Date - Old';
+  groups: (string & {}) | 'none' | 'start' | 'end';
```

Comment:
Unsure what the deal is with `(string & {})`. If you just want to use a "it must be one of these strings" union types then just remove that first bit, unless it's doing something I'm not seeing?

```suggestion
  sortOrder: 'A-Z' | 'Date - New' | 'Date - Old';
  groups: 'none' | 'start' | 'end';
```

Reply:
It keeps it happy when you want to assign one of them from a `string`. It makes it not for real typed but you get the nice reference/autocomplete. I probably wouldn't have bothered except the React types use it as well.

Reply:
The solution to that is probably to make actual types here and use them everywhere.

```ts
export type SortOrder = 'A-Z' | 'Date - New' | 'Date - Old';

export interface SortRules {
  sortOrder: SortOrder;
}
```